Add notion api package support (#2)
ballerina-platform/module-ballerinax-notion: README.md

[![Build](https://github.com/ballerina-platform/module-ballerinax-notion/actions/workflows/ci.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-notion/actions/workflows/ci.yml)
[![Trivy](https://github.com/ballerina-platform/module-ballerinax-notion/actions/workflows/trivy-scan.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-notion/actions/workflows/trivy-scan.yml)

> [![GraalVM Check](https://github.com/ballerina-platform/module-ballerinax-notion/actions/workflows/build-with-bal-test-native.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-notion/actions/workflows/build-with-bal-test-native.yml)

Reviewer: ```suggestion
[![GraalVM Check](https://github.com/ballerina-platform/module-ballerinax-notion/actions/workflows/build-with-bal-test-graalvm.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-notion/actions/workflows/build-with-bal-test-graalvm.yml)
```

Reviewer: Is there any reason for this change?

> [![GraalVM Check](https://github.com/ballerina-platform/module-ballerinax-notion/actions/workflows/build-with-bal-test-native.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-notion/actions/workflows/build-with-bal-test-native.yml)

Reviewer 2: Is this workflow available?

Author: I actually had initially started with what I found [here](https://github.com/ballerina-platform/ballerina-library/blob/main/library-templates/generated-connector-template/files/README.md) will change this back.

3. Provide a name for your integration, and choose the workspace that this integration will have access to.

![integration created](https://github.com/ballerina-platform/module-ballerinax-notion/blob/main/docs/assets/integration-created.png?raw=true)

4. After creating the integration, you will see a page with your Integration Token (which you’ll need in the following steps). Copy the Integration Token and store it securely.

![copy secrets](https://github.com/ballerina-platform/module-ballerinax-notion/blob/main/docs/assets/secret.png?raw=true)

Step 2: Share a Notion Page with Your Integration

1. Go to the Notion page or database that you want your integration to access.
2. Click on the Share button at the top right corner of the page.
3. In the sharing options, search for your integration by the name you provided when creating it.
4. Grant the integration access by selecting it. This allows your integration to read or update the contents of the page or database.

![connect integration](https://github.com/ballerina-platform/module-ballerinax-notion/blob/main/docs/assets/connect-page.png?raw=true)

## Quickstart

### Step 1: Import the module

> Import the `notion` module

Reviewer 3: ```suggestion
Import the `ballerinax/notion` module
```

```bal
import ballerinax/notion;
```

### Step 2: Instantiate a new connector

Create a `notion:ConnectionConfig` with the obtained OAuth2.0 tokens and initialize the connector with it.

```bal
configurable string authToken = ?;

final notion:Client notion = check new notion:Client(
    config = {
        auth: {
            token: authToken
        }
    }
);
```

### Step 3: Invoke the connector operation

Now, utilize the available connector operations.

#### Retrieve an existing page

```bal

> PageResponse result = check notion->/v1/pages/["xyz-page-id"];

Reviewer 3: ```suggestion
PageResponse page = check notion->/v1/pages/["xyz-page-id"];
```

```

#### Creating a page inside another page

```bal
PageBodyParams payload = {
    "parent": {
        "type": "page_id",
        "page_id": testPageId
    },
    "properties": {
        "title": [
        {
            "text": {
            "content": "Test page"
            }
        }
        ]
    },
    "icon": {
        "emoji": "🥬"
    }
};
PageResponse response = check notion->/v1/pages.post(payload);
```

## Examples

> The `ballerinax/notion` connector provides practical examples illustrating usage in various scenarios. Explore these [examples](https://github.com/Jaagrav/module-ballerinax-notion/tree/main/examples), covering use cases like creating pages, retrieving users, and managing databases.

Reviewer: ```suggestion
The `ballerinax/notion` connector provides practical examples illustrating usage in various scenarios. Explore these [examples](https://github.com/ballerina-platform/module-ballerinax-notion/tree/main/examples), covering use cases like creating pages, retrieving users, and managing databases.

1. [Creating a new page](https://github.com/ballerina-platform/module-ballerinax-notion/blob/main/examples/CreateNewPage)
2. [Retrieve existing pages](https://github.com/ballerina-platform/module-ballerinax-notion/blob/main/examples/RetrievePage)
```

## Overview

> [//]: # "TODO: Add overview mentioning the purpose of the module, supported REST API versions, and other high-level details."

Reviewer: Shall we remove the `TODO` comments if you have already fixed them?

3. Provide a name for your integration, and choose the workspace that this integration will have access to.

![integration created](https://github.com/Jaagrav/module-ballerinax-notion/blob/main/docs/assets/integration-created.png?raw=true)

4. After creating the integration, you will see a page with your Integration Token (which you’ll need in the following steps). Copy the Integration Token and store it securely.

![copy secrets](https://github.com/Jaagrav/module-ballerinax-notion/blob/main/docs/assets/secret.png?raw=true)

Step 2: Share a Notion Page with Your Integration

1. Go to the Notion page or database that you want your integration to access.
2. Click on the Share button at the top right corner of the page.
3. In the sharing options, search for your integration by the name you provided when creating it.
4. Grant the integration access by selecting it. This allows your integration to read or update the contents of the page or database.

![connect integration](https://github.com/Jaagrav/module-ballerinax-notion/blob/main/docs/assets/connect-page.png?raw=true)

> ## Quickstart

Reviewer: Shall we follow the same pattern for `quickstart` as here
https://github.com/ballerina-platform/module-ballerinax-googleapis.gmail/blob/master/README.md#quickstart

[//]: # "TODO: Add a quickstart guide to demonstrate a basic functionality of the module, including sample code snippets."

Simply setup the API client and start using it after initialising with the Notion API key

```bal
configurable string authToken = os:getEnv("NOTION_AUTH_TOKEN");

notion:ConnectionConfig config = {
    auth: {
        token: authToken
    }
};

final notion:Client notion = check new (config);
```

> ## Examples

Reviewer: Here, the standard way is to mention the examples you have added in the examples directory.
You can follow this to get an idea.
https://github.com/ballerina-platform/module-ballerinax-googleapis.gmail/blob/master/README.md
